Terminal.Gui/Input/Mouse.cs

```diff
@@ -143,6 +143,11 @@ public class MouseEvent
     /// </remarks>
     public Point ScreenPosition { get; set; }
 
+    /// <summary>
+    ///     Indicates if the current mouse event has first pressed <see langword="true"/>, latest released <see langword="false"/> or none <see langword="null"/>.
+    /// </summary>
+    public bool? IsMouseDown { get; set; }
```

Comment:
See my comments above about the words "Pressed" and "Down". 

In addition, I find it confusing that some mouse state is stored in `MouseEvent` and some is stored in `MouseEvent.Flags`. 

I don't have clarity myself on how we decide, but we should decide explicitly.

Reply:
I decided to create a separate property to avoid with more confusion with flags. Pressed was always kept press. It's only a semantic matter.

Reply:
> It's only a semantic matter.

Semantics are actually the MOST important part of getting clarity on an issue. 

I feel VERY strongly about this. 

Please read this:  

https://people.howstuffworks.com/semantics.htm

Reply:
The `MouseDown` and the `MouseUp` are what you proposed for when the mouse button is press at first time and when the mouse is release at end, respectively. I'm only am using the internal fields for that. Do you prefer I create new Flags for them? In my opinion is duplicating flags because if the `IsMouseDown` is true (`MouseDown`) or false (`MouseUp`) the user can get the right button by inspecting the `ButtonPressed` and the `ButtonReleased`. You know I'm not very good with documentation and any help with this is appreciated. So, for me is much more easy you to say what you want I change in code.

Reply:
You are talking about refactor almost everything and how it will to work later. I only used the existent flags and get them work with Down, Up and some more existent bugs .If you want me to read your famous brain with its brilliant ideas, forget it. If you want to go your own way, feel free and you can close this PR. It's one more.

Reply:
> You are talking about refactor almost everything and how it will to work later.

Yes, I am taking about refactoring at the core. Why? 

Because we know the existing mouse API is has serious problems. For example:

- #3209
- #3284

I am pushing for fixing these things so they are as perfect as they can be. 

I am resisting hacking/patching/workarounds.

Reply:
I know I'm not a perfectionist but you could take what's good about this PR and improve it even more. An API is not created all at once, but rather through constant corrections and improvements. Sometimes it's a mistake to do everything at once.

Reply:
> I know I'm not a perfectionist but you could take what's good about this PR and improve it even more. An API is not created all at once, but rather through constant corrections and improvements. Sometimes it's a mistake to do everything at once.

I told you above that I didn't have the time to do more than a quick code review.

Reply:
> I told you above that I didn't have the time to do more than a quick code review.

I would say when you have time available. In your quick review you may miss something that you don't immediately understand, as you mentioned that I was using static variables in the `ConsoleDriver` but it is in the `Application`. Only after you properly analyze the code tell me what this PR really does not meet what you initially intended. I'm not referring to the rules of good coding which I know I'm not an expert, but in relation to mouse down, mouse up and mouse move. Thanks.

Reply:
>  but in relation to mouse down, mouse up and mouse move. 

Do you want me to code the example View I described above and submit it as a PR? Because that is how I would actually dive deep into this.

Reply:
> You misread my comments. I was suggesting that some of these data items that are statics on View in this PR could make sense to be fields on ConsoleDriver.

But I'm not using static variables in the `View` in this PR but in the `Application`. Am I missing something? If so, I apologize. With age, neurons no longer work well ;-)

Reply:
> Do you want me to code the example View I described above and submit it as a PR? Because that is how I would actually dive deep into this.

Of course. That's how I like to work. If you find that something in my PR is really useful but you want to modify something, I would just appreciate it if you sent me PR for my branch and so, as a team, we can resolve what is not right, improving it.

Reply:
It may not be at the correct standards or what you intended, but I think this PR substantially improves mouse processing.